### PR TITLE
Attach usb controller into right pci endpoint

### PR DESCRIPTION
**Pull request by ljmen**
set the suitable addr for usb controller, to attach usb controller into right pci endpoint

Singed-off-by: ljmen <lmen@redhat.com>

### CODE REVIEW

**Comment by ljmen** [comment]
[root@localhost ~]# avocado run --vt-type libvirt type_specific.io-github-autotest-libvirt.usb_device
/opt/code/avocado/avocado/plugins/run.py:286: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 8b89391aa67950bc8ff513f43ddaae06ebcde168
JOB LOG    : /root/avocado/job-results/job-2020-06-17T03.38-8b89391/job.log
 (1/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pci-root: PASS (98.71 s)
 (2/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pci-root: PASS (100.03 s)
 (3/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pci-root: PASS (103.86 s)
 (4/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pci-root: PASS (87.65 s)
 (5/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pci-root: PASS (89.04 s)
 (6/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pci-root: PASS (91.11 s)
 (7/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb_all.pci-root: PASS (96.74 s)
 (8/9) type_specific.io-github-autotest-libvirt.usb_device.redirdev.spicevmc.usb2_0.companion.pci-root: PASS (48.17 s)
 (9/9) type_specific.io-github-autotest-libvirt.usb_device.redirdev.tcp.usb2_0.companion.pci-root: PASS (165.27 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 882.38 s





[root@localhost ~]# avocado run --vt-type libvirt --vt-machine-type q35   type_specific.io-github-autotest-libvirt.usb_device
/opt/code/avocado/avocado/plugins/run.py:286: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 4fb908f57f7bdc838fd5cd0506db5a68a69d425c
JOB LOG    : /root/avocado/job-results/job-2020-06-17T04.40-4fb908f/job.log
 (01/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (96.95 s)
 (02/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (101.52 s)
 (03/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.dmi-to-pci-bridge: PASS (102.04 s)
 (04/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (100.74 s)
 (05/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (100.53 s)
 (06/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.dmi-to-pci-bridge: PASS (102.31 s)
 (07/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (104.83 s)
 (08/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (104.91 s)
 (09/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.dmi-to-pci-bridge: PASS (104.65 s)
 (10/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (89.01 s)
 (11/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (89.08 s)
 (12/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.dmi-to-pci-bridge: PASS (90.53 s)
 (13/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (89.02 s)
 (14/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (88.56 s)
 (15/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.dmi-to-pci-bridge: PASS (89.58 s)
 (16/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (93.62 s)
 (17/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (94.84 s)
 (18/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.dmi-to-pci-bridge: PASS (93.50 s)
 (19/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb_all.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (96.24 s)
 (20/21) type_specific.io-github-autotest-libvirt.usb_device.redirdev.spicevmc.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (57.23 s)
 (21/21) type_specific.io-github-autotest-libvirt.usb_device.redirdev.tcp.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (167.03 s)
RESULTS    : PASS 21 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 2058.92 s

**Comment by chunfuwen** [comment]
@ljmen ,please resolve travis CI issues

**Comment by ljmen** [comment]
@chunfuwen  done

File: libvirt/tests/src/usb_device.py [hunk 1]
```diff
@@ -227,6 +227,10 @@ def check_alias(device_alias):
             pci_bridge.type = "pci"
             pci_bridge.model = model
             vmxml.add_device(pci_bridge)
+        # find the pci endpoint's name that usb controller will attach
```
Comment (chunfuwen) [inline libvirt/tests/src/usb_device.py:230]: find to Find

File: libvirt/tests/src/usb_device.py [hunk 2]
```diff
@@ -243,6 +247,15 @@ def check_alias(device_alias):
                 if "ich9" not in model:
                     controller.index = i
                 controller.alias = alias
+            # for 'usb_all' case, will not set addr
+            if len(usb_model.split(",")) < 5:
+                ctrl_addr_dict = {'type': 'pci', 'domain': '0x0000', 'bus': '0x0'+str(pci_index_for_usb_controller), 'slot': '0x01', 'function': '0x00'}
```
Comment (chunfuwen) [inline libvirt/tests/src/usb_device.py:252]: can those data be put into cfg file ?

File: libvirt/tests/src/usb_device.py [hunk 3]
```diff
@@ -243,6 +247,15 @@ def check_alias(device_alias):
                 if "ich9" not in model:
                     controller.index = i
                 controller.alias = alias
+            # for 'usb_all' case, will not set addr
+            if len(usb_model.split(",")) < 5:
```
Comment (brucepatricklee) [inline libvirt/tests/src/usb_device.py:251]: maybe we can use a variable to define whether we set addr for controller such as , by default set_addr = yes, but in usb_all test, we set set_addr = false

**Comment by ljmen** [comment]
[root@localhost cfg]#  avocado run  --vt-type libvirt type_specific.io-github-autotest-libvirt.usb_device
/opt/code/avocado/avocado/plugins/run.py:286: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 410aa8f976baf2db6dd80ce10da5b267c1bf5572
JOB LOG    : /root/avocado/job-results/job-2020-07-01T05.44-410aa8f/job.log
 (1/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pci-root: PASS (95.30 s)
 (2/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pci-root: PASS (117.11 s)
 (3/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pci-root: PASS (103.54 s)
 (4/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pci-root: PASS (88.01 s)
 (5/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pci-root: PASS (89.09 s)
 (6/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pci-root: PASS (92.03 s)
 (7/9) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb_all.pci-root: PASS (93.70 s)
 (8/9) type_specific.io-github-autotest-libvirt.usb_device.redirdev.spicevmc.usb2_0.companion.pci-root: PASS (47.71 s)
 (9/9) type_specific.io-github-autotest-libvirt.usb_device.redirdev.tcp.usb2_0.companion.pci-root: PASS (165.73 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 893.95 s




[root@localhost cfg]#  avocado run  --vt-type libvirt --vt-machine-type q35  type_specific.io-github-autotest-libvirt.usb_device
/opt/code/avocado/avocado/plugins/run.py:286: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : b3dd288e4f0bcaabe3feea0588d7de17fa064771
JOB LOG    : /root/avocado/job-results/job-2020-07-01T06.34-b3dd288/job.log
 (01/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (100.58 s)
 (02/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (101.80 s)
 (03/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.nec_xhci_hub.pcie-root.dmi-to-pci-bridge: PASS (101.40 s)
 (04/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (100.98 s)
 (05/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (100.13 s)
 (06/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb3_0.qemu_xhci.pcie-root.dmi-to-pci-bridge: PASS (100.45 s)
 (07/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (104.57 s)
 (08/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (103.95 s)
 (09/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.vid_pid.usb2_0.companion.pcie-root.dmi-to-pci-bridge: PASS (103.81 s)
 (10/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (88.24 s)
 (11/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (90.07 s)
 (12/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.nec_xhci_hub.pcie-root.dmi-to-pci-bridge: PASS (88.04 s)
 (13/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (90.07 s)
 (14/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (88.05 s)
 (15/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb3_0.qemu_xhci.pcie-root.dmi-to-pci-bridge: PASS (87.98 s)
 (16/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (95.72 s)
 (17/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.pcie-root-port.pcie-switch-upstream-port: PASS (93.58 s)
 (18/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb2_0.companion.pcie-root.dmi-to-pci-bridge: PASS (92.69 s)
 (19/21) type_specific.io-github-autotest-libvirt.usb_device.passthrough.bus_dev.usb_all.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (95.17 s)
 (20/21) type_specific.io-github-autotest-libvirt.usb_device.redirdev.spicevmc.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (49.21 s)
 (21/21) type_specific.io-github-autotest-libvirt.usb_device.redirdev.tcp.usb2_0.companion.pcie-root.pcie-root-port.pcie-to-pci-bridge: PASS (165.36 s)
RESULTS    : PASS 21 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 2044.13 s

**Comment by ljmen** [comment]
@brucepatricklee @chunfuwen   finished to update,thanks for your review

**Pull request merged**